Upstream bug: https://bugzilla.mozilla.org/attachment.cgi?id=321598

We've had an issue on debian with the mips xpcom stubs because gcc 4.3 changed
its default from PIC to non-PIC. While the whole c++ code is built as PIC code,
the xptc stubs assembly ends up built as non-PIC.

It would be safer to build the assembly with -fPIC when C++ is built with -fPIC,
to avoid future possible problems with gcc changing its default again on
some random architecture. The attached patch does it for linux, but it might be
necessary to do the same on other architectures.

Patch $Revision: 1.1 $
diff --git a/configure.in b/configure.in
index 3730f0e..516da66 100644
--- a/configure.in
+++ b/configure.in
@@ -1279,6 +1279,7 @@ if test "$GNU_CC"; then
     WARNINGS_AS_ERRORS='-Werror'
     DSO_CFLAGS=''
     DSO_PIC_CFLAGS='-fPIC'
+    ASFLAGS="$ASFLAGS -fPIC"
     _MOZ_RTTI_FLAGS_ON=${_COMPILER_PREFIX}-frtti
     _MOZ_RTTI_FLAGS_OFF=${_COMPILER_PREFIX}-fno-rtti
     _MOZ_EXCEPTIONS_FLAGS_ON='-fhandle-exceptions'
